src/wallet/scriptpubkeyman.cpp

```diff
@@ -2258,13 +2258,20 @@ const std::vector<CScript> DescriptorScriptPubKeyMan::GetScriptPubKeys() const
     return script_pub_keys;
 }
 
-bool DescriptorScriptPubKeyMan::GetDescriptorString(std::string& out) const
+bool DescriptorScriptPubKeyMan::GetDescriptorString(std::string& out, const bool priv) const
 {
     LOCK(cs_desc_man);
 
     FlatSigningProvider provider;
     provider.keys = GetKeys();
 
+    if (priv) {
+        // For private version always return master key to avoid exposing child private keys.
+        // The risk implications of exposing child private key together with parent xpub
+        // could be not obvious for users.
```

Comment:
```diff
-        // For private version always return master key to avoid exposing child private keys.
-        // The risk implications of exposing child private key together with parent xpub
-        // could be not obvious for users.
+        // For the private version, always return the master key to avoid
+        // exposing child private keys. The risk implications of exposing child
+        // private keys together with the parent xpub may be non-obvious for users.
```